14/umbraco-cms/fundamentals/code/subscribing-to-notifications.md

```diff
@@ -1,20 +1,30 @@
 ---
 description: >-
-  Subscribing to notifications allows you to execute custom code on a number of
-  operations both before and after the operation occurs
+  Subscribing to notifications allows you to listen to specific events and run custom code in response.
 ---
 
 # Subscribing To Notifications
 
-Subscribing to notifications allows you to execute custom code on a number of operations both before and after the operation occurs. All you need to follow this guide is an Umbraco installation with some content, for example the Umbraco starter kit.
+Subscribing to notifications allows you to run custom code in response to specific events, such as when the content is created, updated, or deleted. This feature enables you to automate tasks, validate data, log actions, and implement other custom functionalities to enhance your content management system.
 
-## Subscribing to a notification
+To follow this guide, ensure you have an Umbraco installation with content, such as the Umbraco starter kit. In this article, we will walk you through the process of logging a message every time a document is published in Umbraco.
 
-Let us add a string of text to the log when a document is published. The log is useful for debugging. Different parts of the Umbraco codebase 'log' key events, warnings and errors to the log.
+## Create a Notification Handler
 
-We react to notifications in Umbraco inside a notification handler, so let's create one. Add a new C# class to our project - call it _LogWhenPublishedHandler_ and use `: INotificationHandler<ContentPublishedNotification>` to identify our code as a handler, that will handle `ContentPublishedNotification`. We'll need to add `using Umbraco.Cms.Core.Events;` to the top of the .cs file. Because the notifications that you can subscribe to are found in the core notifications namespace, we also need to add a using statement for that: `using Umbraco.Cms.Core.Notifications;`.
+We will add a string of text to the log whenever a document is published. This log is useful for debugging, as different parts of the Umbraco codebase log key events, warnings, and errors.
 
-We now have a class that looks like this:
+1. Add a new C# class to your project. For example: **LogWhenPublishedHandler**.
+2. Implement the `INotificationHandler<ContentPublishedNotification>` interface to identify this class as a handler for content publication events.
+3. Add the necessary `using` statements at the top of your file:
+
+    ```csharp
+    using Umbraco.Cms.Core.Events;
+    using Umbraco.Cms.Core.Notifications;
+    ```
+
+Your class should now look like this:
+
+{% code title="LogWhenPublishedHandler.cs" overflow="wrap" lineNumbers="true" %}
```

Comment:
I don't think we usually use the line numbers or the overflow.

```suggestion
{% code title="LogWhenPublishedHandler.cs"%}
```

Reply:
We have used line numbers in some other articles. Although we haven't used overflow, adding it formats the block to prevent it from running through or overflowing the code block stylesheet. Essentially, it styles the code to display neatly. I can remove it, but the code block looks better with it 😊

Reply:
Sure! As long as it's consistent I'm happy 💪